### PR TITLE
Feature/justiceotuya/connecting redux

### PR DESCRIPTION
**Pull request by justiceotuya**

### CODE REVIEW

File: components/timeLine/reducers.js [hunk 1]
```diff
@@ -17,30 +16,71 @@ export const reducers = (state = initialState, action) => {
...
     case FETCH_PROFILE_REQUEST:
         return { ...state, isFetching: true };
+
     case FETCH_PROFILE_DATA_SUCCESS:
         return { ...state, isFetching: false, timelineData };
+
     case FETCH_PROFILE_DATA_FAILURE:
         return { ...state, error, isFetching: false };
+
     case TOGGLE_MODAL:
```
Comment (EngrEric) [inline components/timeLine/reducers.js]: I think controlling the modal will be better in the particular component you need it.
i.e setting an internal state instead. number of code will reduce and infinitesimal good performance will be achieved

File: components/timeLine/components/TimeLine.jsx [hunk 2]
```diff
@@ -179,16 +154,18 @@ class TimeLine extends React.Component {
...
+                                profileData={timelineData}
                                 handleLikeButton={this.handleLikeButton}
-                                handleOk={this.handleOk}
-                                handleOnChange={this.handleCommentValue}
-                                textValue={commentValue}
+                                handleFavButton={this.handleFavButton}
+                                //   liked={this.handleLikeButton}
+                                //   likeCount={likeCount}
+                                //   activeComment={activeComment}
+                                // activeLikeButton={timelineData.like}
```
Comment (EngrEric) [inline components/timeLine/components/TimeLine.jsx]: remove unused codes

**Review by EngrEric — COMMENTED** [review]
Looks good, just minor changes

**Review by olajohn-ajiboye — DISMISSED** [review]

File: components/timeLine/reducers.js (hunk 1 above)
Comment (Way2nnadi) [reply components/timeLine/reducers.js]: i would second that

File: components/timeLine/index.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { reducers } from './reducers';
-import * as components from './components';
+import * as components from './components/index';
```
Comment (Way2nnadi) [inline components/timeLine/index.js]: no need for `/index`, node imports from index by default

File: components/timeLine/constants.js [hunk 4]
```diff
@@ -1,3 +1,16 @@
+export const data = (id, post) => ({
```
Comment (Way2nnadi) [inline components/timeLine/constants.js]: this is not a constant variable, this is a function. do not put it in the `constants.js` file. You can create a `utils` file if you need or define it in the function that uses this if it is not reuseable

File: components/timeLine/components/TimeLine.jsx [hunk 5]
```diff
@@ -179,16 +154,18 @@ class TimeLine extends React.Component {
...
+                                handleComment={this.handleComment}
+                                // handleComment={this.handleComment}
+                            //   handleCreateStatus={this.handleCreateStatus}
+                            //   handleOnChange={this.handleCommentValue}
+                            //   textValue={commentValue}
```
Comment (Way2nnadi) [inline components/timeLine/components/TimeLine.jsx]: @justiceotuya  clean up this file

File: components/timeLine/components/TimeLine.css [hunk 6]
```diff
@@ -192,6 +203,14 @@
   text-align: justify;
 }
 
+.ant-list-item.ant-list-item-no-flex{
+  display: grid;
+}
+
+.ant-list-item-action{
+    grid-row: 4;
```
Comment (Way2nnadi) [inline components/timeLine/components/TimeLine.css]: keep css formatting consistent,
```
.ant-list-item-action {
    grid-row: 4;
}
```

File: components/timeLine/actions.js [hunk 7]
```diff
@@ -40,12 +39,38 @@ export const fetchProfileData = () => dispatch => {
...
+            comment,
+            email,
+            favouriteCount,
+            favourited,
+            firstName,
+            id,
+            lastName,
+            liked,
+            likes,
+            post,
```
Comment (Way2nnadi) [inline components/timeLine/actions.js]:
```
export const handlePostUpdate = payload => ({
    payload,
    type: ADD_POST_TO_TIMELINE,
};
```
handle the payload destructuring in the `reducer`

File: components/timeLine/actions.js [hunk 8]
```diff
@@ -1,13 +1,10 @@
 import actionTypes from './actionTypes';
 
 const {
-    TOGGLE_MODAL, FETCH_PROFILE_REQUEST,
-    FETCH_PROFILE_DATA_FAILURE,
-    FETCH_PROFILE_DATA_SUCCESS, UPDATE_STATUS, ADD_POST_TO_TIMELINE,
+    FETCH_PROFILE_REQUEST, FETCH_PROFILE_DATA_FAILURE, FETCH_PROFILE_DATA_SUCCESS,
+    UPDATE_STATUS, ADD_POST_TO_TIMELINE, TOGGLE_POST_LIKE, TOGGLE_POST_FAV, TOGGLE_COMMENT_BUTTON,
```
Comment (Way2nnadi) [inline components/timeLine/actions.js]: all on one line or each on there own line

File: components/timeLine/actions.js [hunk 9]
```diff
@@ -38,14 +35,22 @@ export const fetchProfileData = () => dispatch => {
...
+
+export const favButtonClicked = id => ({
+    payload: id,
+    type: TOGGLE_POST_FAV,
+});
+
+export const commentButtonClicked = id => ({
+    payload: id,
+    type: TOGGLE_COMMENT_BUTTON,
+});
```
Comment (Way2nnadi) [inline components/timeLine/actions.js:62]: the pattern for each one of these action creators should be `payload` as the parameter.
```
export const commentButtonClicked = payload => ({
    payload,
    type: TOGGLE_COMMENT_BUTTON,
});
```
Make sure that when you call the action creator to pass it an object
```
...
commentButtonClicked({ id });
...
```

File: components/timeLine/components/TimeLine.css [hunk 10]
```diff
@@ -192,6 +203,14 @@
   text-align: justify;
 }
 
+.ant-list-item.ant-list-item-no-flex {
+  display: grid;
+}
+
+.ant-list-item-action {
+    grid-row: 4;
+}
```
Comment (Way2nnadi) [inline components/timeLine/components/TimeLine.css:216]: formatting is still not consistent on this page

File: components/timeLine/components/TimeLine.jsx [hunk 11]
```diff
@@ -1,142 +1,147 @@
...
-import TimeLinePopularTopic from './TimeLinePopularTopic';
+import CreatePostModal from './CreatePostModal';
+import { getStatusValue, getTimelineData } from '../selectors';
 import TimeLinePosts from './TimeLinePosts';
-import TimeLineProfileInfo from './TimeLineProfileInfo';
+import { STRINGS } from '../constants';
 
 const { CREATE_POST_PLACEHOLDER, TIMELINE_TITLE } = STRINGS;
 const { PageLayout } = components;
+export const data = (id, post) => ({
```
Comment (Way2nnadi) [inline components/timeLine/components/TimeLine.jsx:25]: `const generateData`

File: components/timeLine/reducers.js [hunk 12]
```diff
@@ -7,40 +7,76 @@ const initialState = {
...
+
+    case TOGGLE_POST_LIKE:
+        state.timelineData.map(item => {
+            const { id, liked, likes } = item;
+            if (id === payload) {
+                item.liked = !liked;
+                item.likes = liked ? likes - 1 : likes + 1;
+            }
+            return { liked, likes };
+        });
```
Comment (Way2nnadi) [inline components/timeLine/reducers.js:57]: do not update the state object directly

File: components/timeLine/reducers.js [hunk 13]
```diff
@@ -7,40 +7,76 @@ const initialState = {
...
+
+    case TOGGLE_POST_FAV:
+        state.timelineData.map(item => {
+            const { id, favourited, favouriteCount } = item;
+            if (id === payload) {
+                item.favourited = !favourited;
+                item.favouriteCount = favourited ? favouriteCount - 1 : favouriteCount + 1;
+            }
+            return { favouriteCount, favourited };
+        });
```
Comment (Way2nnadi) [inline components/timeLine/reducers.js:71]: do not update the state object directly

File: components/timeLine/reducers.js [hunk 14]
```diff
@@ -7,40 +7,76 @@ const initialState = {
...
+        return { ...state, timelineData: [...state.timelineData] };
+
+    case TOGGLE_COMMENT_BUTTON:
+        state.timelineData.map(item => {
+            const { id, isCommentOpen } = item;
+            if (id === payload) {
+                item.isCommentOpen = !isCommentOpen;
+            }
+            return isCommentOpen;
+        });
```
Comment (Way2nnadi) [inline components/timeLine/reducers.js:81]: do not update the state object directly. Save the map over a new copy of `state.timelineData`

File: components/timeLine/components/TimeLine.jsx [hunk 15]
```diff
@@ -1,142 +1,147 @@
...
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
 import { Divider, Icon } from 'antd';
+import PropTypes from 'prop-types';
 import React from 'react';
 
+import {
+    commentButtonClicked, favButtonClicked, fetchProfileData, handlePostUpdate,
+    likeButtonClicked, setPostUpdateField
+} from '../actions';
```
Comment (EngrEric) [inline components/timeLine/components/TimeLine.jsx:15]:
```
import * as timeLineActions  from '../actions';
```

File: components/timeLine/components/TimeLine.jsx [hunk 16]
```diff
@@ -1,142 +1,147 @@
...
+            handlePostUpdate, setPostUpdateField, statusValue, timelineData,
+        } = this.props;
+
+        const { isModalOpen } = this.state;
+
+        // get post
+        handlePostUpdate(data(timelineData.length + 1, statusValue));
+
+        // close modal
+        if (isModalOpen) this.modalHandler();
```
Comment (EngrEric) [inline components/timeLine/components/TimeLine.jsx]:
```
if(isModalOpen) {
   this.modalHandler();
}
```
OR 
`isModalOpen && this.modalHandler()`

File: components/timeLine/components/TimeLine.jsx [hunk 17]
```diff
@@ -197,4 +188,38 @@ class TimeLine extends React.Component {
...
     }
 }
-export default TimeLine;
+
+const mapStateToProps = state => ({
+    statusValue: getStatusValue(state),
+    timelineData: getTimelineData(state),
+});
+
+const mapDispatchToProps = dispatch => bindActionCreators({
```
Comment (EngrEric) [inline components/timeLine/components/TimeLine.jsx]: `const mapDispatchToProps = dispatch => bindActionCreators(timeLineActions, dispatch);`

File: components/timeLine/components/TimeLinePosts.jsx [hunk 18]
```diff
@@ -11,117 +11,150 @@ import './TimeLine.css';
...
-                        paragraph={paragraph}
-                        title={title}
-                        loading={loading}
-                        active={active}
-                        avatar={avatar}
-                        className="skeleton-section"
-                    />
+                            ) : <div />
+                        }
+                        {post.substr(0, 150)}
```
Comment (EngrEric) [inline components/timeLine/components/TimeLinePosts.jsx]: use `substring()` instead

>`substr()` is considered a legacy function and should be avoided when possible. It is not part of the core JavaScript language and may be removed in the future

File: components/timeLine/reducers.js [hunk 19]
```diff
@@ -7,40 +7,78 @@ const initialState = {
     statusValue: '',
     timelineData: [],
 };
-
 export const reducers = (state = initialState, action) => {
```
Comment (EngrEric) [inline components/timeLine/reducers.js]: this is more neat hence you aren't exporting any other thing here
`export default (state = initialState, action) => {`

File: components/timeLine/reducers.js [hunk 20]
```diff
@@ -7,40 +7,78 @@ const initialState = {
...
     const {
         CLEAR_STATUS_FIELD,
         FETCH_PROFILE_REQUEST,
         FETCH_PROFILE_DATA_FAILURE,
         FETCH_PROFILE_DATA_SUCCESS,
-        TOGGLE_MODAL,
         UPDATE_STATUS,
         ADD_POST_TO_TIMELINE,
+        TOGGLE_POST_LIKE,
+        TOGGLE_POST_FAV, TOGGLE_COMMENT_BUTTON,
```
Comment (EngrEric) [inline components/timeLine/reducers.js]: each should be in a new line

**Review by EngrEric — CHANGES_REQUESTED** [review]
Just need some minor changes

**Review by Way2nnadi — APPROVED** [review]
LGTM

**Pull request merged**